components/data_prefetch/data_prefetch.tsx

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import PQueue from 'p-queue';
+import {Channel} from 'mattermost-redux/types/channels';
+
+import {Constants} from 'utils/constants';
+import {loadProfilesForSidebar} from 'actions/user_actions.jsx';
+
+const queue = new PQueue({concurrency: 2});
+
+type Props = {
+    currentChannelId: string;
+    prefetchQueueObj: Record<string, string[]>;
+    prefetchRequestStatus: Record<string, any>;
+    unreadChannels: Channel[];
+    actions: {
+        prefetchChannelPosts: (channelId: string, delay: number | undefined) => Promise<any>;
+    };
+}
+
+export default class DataPrefetch extends React.PureComponent<Props, {}> {
+    private prefetchTimeout?: number;
+
+    async componentDidUpdate(prevProps: Props) {
+        const {currentChannelId, prefetchQueueObj} = this.props;
+        if (!prevProps.currentChannelId && currentChannelId) {
+            queue.add(async () => this.prefetchPosts(currentChannelId));
+            await loadProfilesForSidebar();
```

Comment:
Because it is hard to guess when the profiles are completely loaded so we can start prefetching with the way existing code works. Moving it here makes it easy to assert for dispatching prefetching actions.

Reply:
And we are loading the profiles on first load on LHS only after current channel is set.

Reply:
This is going to make it so that we make multiple requests to load the profiles, although I can't really see any better way to do that since we don't set a flag or anything to say that profiles have been loaded for the whole sidebar.

Reply:
This is the only place we are loading the profiles for the first time now. I changed it so that global actions does not, Just in case if that wasn't clear

Reply:
Ah, I missed that we weren't calling it elsewhere.